src/ucp/core/ucp_context.c

```diff
@@ -1377,6 +1383,21 @@ static ucs_status_t ucp_fill_config(ucp_context_h context,
     ucs_debug("estimated bcopy bandwidth is %f",
               context->config.ext.bcopy_bw);
 
+    for (proto_id = 0; proto_id < ucp_protocols_count; ++proto_id) {
+        if (config->protos.mode == UCS_CONFIG_ALLOW_LIST_ALLOW_ALL) {
```

Comment:
maybe check `config->protos.mode == UCS_CONFIG_ALLOW_LIST_ALLOW_ALL` outside the loop?